scripts/codegen.js

```diff
@@ -0,0 +1,123 @@
+// @ts-check
+
+const fs = require("fs/promises");
+const path = require("path");
+const telescope = require("@cosmology/telescope").default;
+
+async function codegen() {
+  const cwd = process.cwd();
+  const outPath = path.resolve(cwd, "packages/core/src/codegen/");
+  await fs.rm(outPath, { recursive: true, force: true }).catch(() => {});
+  await telescope({
+    protoDirs: [
+      "./node_modules/@protobufs/",
+      path.resolve(cwd, "vendor/cosmos-proto/proto"),
+      path.resolve(cwd, "vendor/cosmos-sdk/proto"),
+      path.resolve(cwd, "vendor/evmos/proto"),
+      path.resolve(cwd, "vendor/noble-cctp/proto"),
+    ],
+    outPath: path.resolve(cwd, "packages/core/src/codegen/"),
+    options: {
+      aminoEncoding: {
+        customTypes: {
+          useCosmosSDKDec: true,
+        },
+        enabled: true,
+        exceptions: {
```

Comment:
How did we come up with the exceptions and the typeToAmino function? Are we imitating what we've seen work elsewhere or can you explain why this is required?

Reply:
https://github.com/evmos/evmos/blob/v16.0.3/crypto/ethsecp256k1/ethsecp256k1.go#L31-L36

Reply:
Will add comments on codegen script, or even a dev docs explaining it.